torch/distributed/nn/functional.py

```diff
@@ -262,31 +266,34 @@ class _Reduce_Scatter(Function):
     @staticmethod
     def forward(ctx, op, group, tensor, *input_tensor_list):
         ctx.group = group
-        dist.reduce_scatter(tensor, list(input_tensor_list), op=op, group=group)
+        input_tensor_list = [t if t.is_contiguous() else t.contiguous() for t in input_tensor_list]
+        dist.reduce_scatter(tensor, input_tensor_list, op=op, group=group)
         return tensor
 
     @staticmethod
     def backward(ctx, grad_output):
-        return (None, None, None) + _AllGather.apply(
-            ctx.group, grad_output.contiguous()
-        )
+        return (None, None, None) + _AllGather.apply(ctx.group, grad_output)
 
 
 class _AllGather(Function):
     @staticmethod
     def forward(ctx, group, tensor):
         ctx.group = group
         out_tensor_list = [
-            torch.empty_like(tensor) for i in range(dist.get_world_size(group=group))
+            torch.empty_like(tensor) for _ in range(dist.get_world_size(group=group))
         ]
+
+        if not tensor.is_contiguous():
+            tensor = tensor.contiguous()
```

Comment:
nit: this two line thing is basically what contiguous() does anyway.

Reply:
Yes... 
Previously, I thought is_contiguous() is a python-only metadata check, and contiguous() would call into pybind and C++ kernel, so I have fast bypass check.
After the dispatcher lab, I learnt contiguous() op is implemented with manual_cpp_binding, which skips the dispatcher path, and it also have a fast track to bypass.